src/nevermined/api/AssetsApi.ts

```diff
@@ -0,0 +1,464 @@
+        const { files } = attributes.main
+
+        if (!serviceEndpoint) {
+            throw new AssetError(
+                'Consume asset failed, service definition is missing the `serviceEndpoint`.'
+            )
+        }
+
+        this.logger.log('Consuming files')
+
+        resultPath = resultPath
+            ? `${resultPath}/datafile.${ddo.shortId()}.${index}/`
+            : undefined
+
+        await this.nevermined.services.node.consumeService(
+            did,
+            agreementId,
+            serviceEndpoint,
+            consumerAccount,
+            files,
+            resultPath,
+            fileIndex
+        )
+        this.logger.log('Files consumed')
+
+        if (resultPath) {
+            return resultPath
+        }
+        return true
+    }
+
+
+    /**
+     * Returns the owner of an asset.
+     * @param did - Decentralized ID.
+     * @returns The address of the owner of the asset
+     */
+    public async owner(did: string): Promise<string> {
+        const ddo = await this.resolve(did)
+        const checksum = ddo.checksum(didZeroX(did))
+        const { creator, signatureValue } = ddo.proof
+        const signer = await this.nevermined.utils.signature.verifyText(
+            checksum,
+            signatureValue
+        )
+
+        if (signer.toLowerCase() !== creator.toLowerCase()) {
+            this.logger.warn(
+                `Owner of ${ddo.id} doesn't match. Expected ${creator} instead of ${signer}.`
+            )
+        }
+
+        return creator
+    }
+
+    /**
+     * Returns the assets owned by an address
+     * @param owner - The address to check 
+     * @returns List of DIDs owned by the address
+     */
+    public async ownerAssets(owner: string): Promise<string[]> {
+        return this.nevermined.keeper.didRegistry.getAttributesByOwner(owner)
+    }
+
+    /**
+     * Transfer ownership of an asset.
+     * @param did - Asset DID.
+     * @param newOwner - Ethereum address of the new owner of the DID.
+     * @returns Returns ethers transaction receipt.
+     */
+    public async transferOwnership(
+        did: string,
+        newOwner: string,
+        params?: TxParameters
+    ): Promise<ContractReceipt> {
+        const owner = await this.nevermined.assets.owner(did)
+        return this.nevermined.keeper.didRegistry.transferDIDOwnership(
+            did,
+            newOwner,
+            owner,
+            params
+        )
+    }
+
+    /**
+     * Returns the assets of a consumer.
+     * @param consumer - Consumer address.
+     * @returns List of DIDs.
+     */
+    public async consumerAssets(consumer: string): Promise<string[]> {
+        return (
+            await this.nevermined.keeper.conditions.accessCondition.getGrantedDidByConsumer(
+                consumer
+            )
+        ).map(({ did }) => did)
+    }
+
+
+    public async retire(did: string) {
+        return this.nevermined.services.metadata.delete(did)
```

Comment:
need to revisit this. I don't believe this is working since it's a protected endpoint

Reply:
It's not protected in the marketplace api because the service authorize the delete, only allowing the owner of the asset to call that action.